### ISSUE TITLE
Create 'multi-synteny adapter'

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse-components#4724]
Could investigate use of "all vs all" PAF

This is done by e.g.

```
cat hg19.fa hg38.fa t2t.fa hg002.fa > all.fa
minimap2 all.fa all.fa > all_vs_all.paf

```

This would be helpful because currently for N assemblies, you could potentially have to load N^2 pairwise comparison tracks

**Comment by baozg** [comment]
Did this also include genomes / annotation json list?

**Comment by cmdcolin** [comment]
not sure exactly what you mean (feel free to elaborate) but I haven't implemented work on this yet, but I look forward to adding it. 

My current feeling is that I will potentially require that the input fasta files use PanSN naming (https://github.com/pangenome/PanSN-spec) or something like this

**Comment by baozg** [comment]
I think this could two different level visulization. PanSN would be good start, especially with wfmash alignment. It should be fine for dozen of genomes. By the way, current jbrowser did awesome job for multiple genome together, interactively visulization help me understanding what's going on. Another nice way to visulaize them is the way like `PPanG` (https://github.com/SJTU-CGM/PPanG/), which intergrate SequenceTubeMap and Jbrowser2.

But I am worried about performance of multiple synteny visulaization with hunderds of assemblies. Maybe it should design a different way. Due to the complexitiy of multiple genome alignment, even after computation liftover them into one reference cannot see the insertion with MAF ( eg. https://github.com/pangenome/jbrowse-visualization use the pseduomaf -> maf2bed to visulization them into chunk). But large-scale pangenome MSA is very (now I am using ~300 A.thaliana genomes) . Even display them in one screen is very hard. Do you have any plan to support large-scale pangenome visulization? I still perfer the way fro bam visulization or mutliple PAF together for whole genome alignment. Chunking genome into ~50kb segments as reads (https://github.com/igvteam/igv/issues/1551#issuecomment-2331739022) could trick, but I am wondering if Jbrowser could directly chunk the whole genome alignment since reads mapping have more problem in complex region.

**Comment by cmdcolin** [comment]
I had not seen this repo: https://github.com/pangenome/jbrowse-visualization 

This is a great resource

I think we need a multi-pronged attack that is needed to tackle this...or, just multiple options for how to view the data!


## MAF viewer

 I think your MAF files are interesting they contain very very long chunks, maybe created via this pseudo-MAF procedure. This indeed creates a pathologic case where to view even a small region of the genome, the browser has to decode a very large amount of data.

Most MAF I worked with had much smaller chunks, so they are relatively fast to view. 

I am wondering if the pseudoMAF technique could be improved so that it creates the smaller chunks more readily. There is also the TAF format, which i am trying to add support for as well. TAF is actually a tricky format so it is not quite ready (I even added it to the codebase https://github.com/cmdcolin/jbrowse-plugin-mafviewer?tab=readme-ov-file#example-bgziptaffyadapter-config but it had some unexpected bugs so I removed it from the UI)

Regarding that viewing insertions is hard in the MAF, that is indeed true. Potentially the UI could be improved to show the insertions on mouseover. Or a synthetic pseudo-reference sequence that contains all or most of the insertions in the entire pangenome could be created, and the data could be viewed relative to that pseudo- reference. This would involve more data preparation, but if the rendering results are good, it could be nice.


## Graph genome viewer

Some of my recent efforts have included a proof of concept for rendering a graph genome GFA file as a view in JBrowse 2

https://github.com/cmdcolin/jbrowse-plugin-graphgenomeviewer/

This is a very basic implementation and doesn't have any interactivity with the linear genome viewer yet, but I hope that this type of integration can be added. This plugin currently is focused on allowing small-ish GFA, that are for example extracted from a subregion of a graph with the "vg chunk" command. It can't really handle whole genome GFA.


## Tview

I made an experimental plugin called Tview, emulating samtools tview. It uses our multiple sequence alignment viewer (react-msaview) to show reads, and can show basic insertions compared to the reference from reads https://github.com/cmdcolin/jbrowse-plugin-tview

it is mainly focused on reads and might have trouble with very large sequences, and it needs a couple bugs fixed, but might be interesting in some cases

## Multi-variant viewer

We also added the multi-variant viewer from VCF files

https://jbrowse.org/jb2/blog/2024/12/10/v2.18.0-release/#new-multi-variant-display-modes

The multi-variant viewer is focused on SNPs and indels, but I am now trying to think about how it can be adapted to SVs as well


## Multi-copy number viewer

One interesting example of 'multi SV' is this copy-number-variant visualization that I made a demo of for 1000 genome data. I used the multi-wiggle track to show copy number bigwig files across 1kg samples. It includes a little workflow to even "cluster" the data on the fly to show some patterns in the data

![Image](https://github.com/user-attachments/assets/fa2ddd25-559b-484b-a196-64798ea32758)

This clustering of multi-wiggle was just added and will get released in next release https://github.com/GMOD/jbrowse-components/pull/4882

The copy number bigwigs are sourced from https://github.com/KiddLab/kmer_1KG


## Other types of SV in multi-variant renderer

I am also trying to look at other types of SV that could be rendered....thinking about polymorphic mobile element insertions (MEI), inversions, and other types of things. It is tricky because SVs can overlap making it harder to more clearly choose the way that this gets rendered but I think it can be doable. 

## Multi-way synteny

All the above are not using the normal "synteny style" view but tools like like plotsr  (https://github.com/schneebergerlab/plotsr) and mumento (https://github.com/vikshiv/mumemto) have shown me that giant multi-way alignments are indeed possible, we just have to adapt some technique to make it work in jbrowse 2! https://github.com/vikshiv/mumemto

**Comment by cmdcolin** [comment]
also note that a similar idea, with the panel for the graphgenomeviewer, could include a panel for a sequenceTubeMap

**Comment by baozg** [comment]
The whole genome MAF could be chunked with https://github.com/pangenome/mafchunk, that's how we work with jbrowser2.
Compared all "graph-based", I still think we also could looks all SVs just by simple bam, I use IGV with bam by minimap2 or wfmash.  Graph-based seems need more devolepment

- Overview
<img width="1183" alt="Image" src="https://github.com/user-attachments/assets/01f1424f-9f97-452d-9ccf-fae32c59c98a" />

- Per-sample 

<img width="1172" alt="Image" src="https://github.com/user-attachments/assets/72a99bba-1a9e-434a-b699-424bd5d2c293" />

- Well-known inversion in A.thaliana

<img width="979" alt="Image" src="https://github.com/user-attachments/assets/36ef4c96-8dc5-45a9-8d76-73a27d9cdd23" />

**Comment by cmdcolin** [comment]
yes, storing multi-sample assembly-to-assembly alignment in BAM is another way. I am not a huge fan of storing in BAM, because it runs into trouble where you have to potentially decode large amounts of data to just view a small region. PAF has the potential to be better

another thread from earlier this week about assembly-to-assembly alignment in BAM, showing the challenges
https://github.com/GMOD/jbrowse-components/issues/4877

artificially splitting up alignments into 50kb is a workaround (as you have mentioned), but PAF (by virtue of not storing huge amounts of e.g. SEQ data by default) is an alternative. I think representing the true alignment sizes is the ideal. the scalability of PAF has challenges because it is not by default indexable, but the PIF technique that jbrowse adds, where it indexes PAF (bidirectionally, on both query and target) using tabix is a way to try to help with this.

i will do a little more brainstorming. is your data online by chance?

**Comment by baozg** [comment]
That would be great! I totally agree with you that PAF would be more senseable if we only care cigar and corrdinate. How did I load PAF just like BAM not in the synteny view mode. We have lots of assemblies not published yet, but A.thliana also have lots of public data from NCBI. I could make a small subset and send it to you by email. Is the PAF enough for you to play with? Ideally we could load track from query-to-target and query annotation in the same time.

**Comment by cmdcolin** [comment]
sure thing, i'd be happy if you want to send that!

**Comment by cmdcolin** [comment]
We have some new tutorials on our website covering some solutions to this e.g.

ortholog table via MCScan tutorial
https://jbrowse.org/jb2-staging/docs/tutorials/multiway_synteny/

all vs all minimap2 PAF
https://jbrowse.org/jb2-staging/docs/tutorials/allvsall_synteny/

and even some 'pangenome' style approaches are coming online too. check tutorial list here

https://jbrowse.org/jb2-staging/docs/tutorials/

**Comment by cmdcolin** [comment]
could optimistically close for now. it is, to some extent, pretty well solved by those tutorials


please let me know if you have any questions in issues or by email colin.diesh@gmail.com
